src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -52,11 +52,12 @@ def __init__(self, revision, cache_root, client_id, access_token):
 
         if revision is None:
             # Retrieve revision of latest codecov build
-            github_revision = uploader.get_latest_codecov()
-            self.revision = self.githubUtils.get_mercurial(github_revision)
+            self.github_revision = uploader.get_latest_codecov()
+            self.revision = self.githubUtils.get_mercurial(self.github_revision)
             self.from_pulse = False
             suites_to_ignore = []
         else:
+            self.github_revision = self.githubUtils.get_commit(self.revision)
```

Comment:
In this case it's better not to set it as it might fail. The `revision` is not None case is when the task is triggered by a pulse message from a mozilla-central push, and there's a slight delay from when a push is done to when the hg->git mapper is populated. We are generating the zero coverage report only when `revision` is None, so it's ok if we don't set `github_revision` when `revision` is not None.